cli/tools/task.rs

```diff
@@ -55,18 +64,149 @@ pub async fn execute_script(
+    &task_flags.filter
+  {
+    let Some(task_name) = &task_flags.task else {
+      writeln!(&mut std::io::stdout(), "Missing task argument")?;
+      return Ok(0);
+    };
+
+    // Filter based on package name
+    let package_regex = arg_to_regex(filter)?;
+    let task_regex = arg_to_regex(task_name)?;
+
+    let mut packages_task_info: Vec<PackageTaskInfo> = vec![];
+
+    fn matches_package(
+      config: &FolderConfigs,
+      force_use_pkg_json: bool,
+      regex: &Regex,
+    ) -> bool {
+      if !force_use_pkg_json {
+        if let Some(deno_json) = &config.deno_json {
+          if let Some(name) = &deno_json.json.name {
+            if regex.is_match(name) {
+              return true;
+            }
+          }
+        }
+      }
+
+      if let Some(package_json) = &config.pkg_json {
+        if let Some(name) = &package_json.name {
+          if regex.is_match(name) {
+            return true;
+          }
+        }
+      }
+
+      false
+    }
+
+    let workspace = cli_options.workspace();
+    for folder in workspace.config_folders() {
+      if !matches_package(folder.1, force_use_pkg_json, &package_regex) {
+        continue;
+      }
+
+      let member_dir = workspace.resolve_member_dir(folder.0);
+      let mut tasks_config = member_dir.to_tasks_config()?;
+      if force_use_pkg_json {
+        tasks_config = tasks_config.with_only_pkg_json();
+      }
+
+      // Any of the matched tasks could be a child task of another matched
+      // one. Therefore we need to filter these out to ensure that every
+      // task is only run once.
+      let mut matched: HashSet<String> = HashSet::new();
+      let mut visited: HashSet<String> = HashSet::new();
+
+      fn visit_task(
+        tasks_config: &WorkspaceTasksConfig,
+        visited: &mut HashSet<String>,
+        name: &str,
+      ) {
+        if visited.contains(name) {
+          return;
+        }
+
+        visited.insert(name.to_string());
+
+        if let Some((_, TaskOrScript::Task(_, task))) = &tasks_config.task(name)
+        {
+          for dep in &task.dependencies {
+            visit_task(tasks_config, visited, dep);
+          }
+        }
+      }
+
+      // Match tasks in deno.json
+      for name in tasks_config.task_names() {
+        if task_regex.is_match(name) && !visited.contains(name) {
+          matched.insert(name.to_string());
+          visit_task(&tasks_config, &mut visited, name);
+        }
+      }
+
+      packages_task_info.push(PackageTaskInfo {
+        matched_tasks: matched
+          .iter()
+          .map(|s| s.to_string())
+          .collect::<Vec<_>>(),
+        tasks_config,
+      });
+    }
+
+    // Logging every task definition would be too spammy. Pnpm only
+    // logs a simple message too.
+    if packages_task_info
+      .iter()
+      .all(|config| config.matched_tasks.is_empty())
+    {
+      writeln!(
+        &mut std::io::stdout(),
```

Comment:
@marvinhagemeister for the future, we can use `log::warn` instead (which allows people to disable via `--quiet` or `--log-level` and also logs to stderr.